common/src/main/java/com/wynntils/gui/screens/WynntilsDiscoveriesScreen.java

```diff
@@ -0,0 +1,377 @@
+                        poseStack,
+                        I18n.get("screens.wynntils.wynntilsDiscoveries.noDiscoveries"),
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f + 15f,
+                        Texture.QUEST_BOOK_BACKGROUND.width() - 15f,
+                        0,
+                        Texture.QUEST_BOOK_BACKGROUND.height(),
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 30f,
+                        CommonColors.BLACK,
+                        HorizontalAlignment.Center,
+                        VerticalAlignment.Middle,
+                        FontRenderer.TextShadow.NONE);
+    }
+
+    private void renderDescription(PoseStack poseStack, String description, String filterHelper) {
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        description,
+                        20,
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 10,
+                        80,
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 30,
+                        CommonColors.BLACK,
+                        HorizontalAlignment.Left,
+                        FontRenderer.TextShadow.NONE);
+
+        FontRenderer.getInstance()
+                .renderAlignedTextInBox(
+                        poseStack,
+                        filterHelper,
+                        20,
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 10,
+                        105,
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 30,
+                        CommonColors.BLACK,
+                        HorizontalAlignment.Left,
+                        FontRenderer.TextShadow.NONE);
+    }
+
+    @Override
+    protected DiscoveryButton getButtonFromElement(int i) {
+        int offset = i % getElementsPerPage();
+        return new DiscoveryButton(
+                Texture.QUEST_BOOK_BACKGROUND.width() / 2 + 15,
+                offset * 13 + 25,
+                Texture.QUEST_BOOK_BACKGROUND.width() / 2 - 37,
+                9,
+                elements.get(i));
+    }
+
+    @Override
+    protected void reloadElementsList(String searchTerm) {
+        elements.addAll(Stream.concat(
+                        webDiscoveryInfoCache.stream()
+                                .filter(discoveryInfo -> switch (discoveryInfo.getType()) {
+                                    case TERRITORY -> showUndiscoveredTerritory;
+                                    case WORLD -> showUndiscoveredWorld;
+                                    case SECRET -> showUndiscoveredSecrets;
+                                })
+                                .filter(discoveryInfo -> DiscoveryManager.getAllDiscoveries()
+                                        .noneMatch(
+                                                discovery -> discovery.getName().equals(discoveryInfo.getName()))),
```

Comment:
What is this code for?

Reply:
Api has every discovery, we make sure to not display them twice (if we have parsed it from quest book)